src/Components/GenericExpressCheckout/CustomerRegistrationUtil.php

```diff
@@ -28,57 +29,72 @@ public function getCustomerDataBagFromGetCheckoutSessionResponse(array $response
     {
         $salutationId = $this->getSalutationId($context);
 
+        $billingAddress = array_filter([
+            'salutationId' => $salutationId,
+            'company' => $this->extractBillingData($response, 'company'),
+            'firstName' => $this->extractBillingData($response, 'firstname'),
+            'lastName' => $this->extractBillingData($response, 'lastname'),
+            'street' => $this->extractBillingData($response, 'street'),
+            'additionalAddressLine1' => $this->extractBillingData($response, 'addressaddition'),
+            'zipcode' => $this->extractBillingData($response, 'zip'),
+            'city' => $this->extractBillingData($response, 'city'),
+            'countryId' => $this->getCountryIdByCode($this->extractBillingData($response, 'country') ?? '', $context),
+            'phone' => $this->extractBillingData($response, 'telephonenumber'),
+        ]);
+
+        $shippingAddress = array_filter([
+            'salutationId' => $salutationId,
+            'company' => $this->extractShippingData($response, 'company'),
+            'firstName' => $this->extractShippingData($response, 'firstname'),
+            'lastName' => $this->extractShippingData($response, 'lastname'),
+            'street' => $this->extractShippingData($response, 'street'),
+            'additionalAddressLine1' => $this->extractShippingData($response, 'addressaddition'),
+            'zipcode' => $this->extractShippingData($response, 'zip'),
+            'city' => $this->extractShippingData($response, 'city'),
+            'countryId' => $this->getCountryIdByCode($this->extractShippingData($response, 'country') ?? '', $context),
+            'phone' => $this->extractShippingData($response, 'telephonenumber'),
+        ]);
+
+        $isBillingAddressComplete = $this->hasAddressRequiredData($billingAddress);
+        $isShippingAddressComplete = $this->hasAddressRequiredData($shippingAddress);
+
+        if (!$isBillingAddressComplete && !$isShippingAddressComplete) {
+            throw new RuntimeException($this->translator->trans('PayonePayment.errorMessages.genericError'));
+        }
+
+        if (!$isBillingAddressComplete && $isShippingAddressComplete) {
+            $billingAddress = $shippingAddress;
+        }
+
         $customerData = new RequestDataBag([
             'guest' => true,
             'salutationId' => $salutationId,
             'email' => $response['addpaydata']['email'],
-            'firstName' => $this->extractBillingData($response, 'firstname'),
-            'lastName' => $this->extractBillingData($response, 'lastname'),
+            'firstName' => $billingAddress['firstName'], /** @phpstan-ignore offsetAccess.notFound */
```

Comment:
Due to the function “hasAddressRequiredData” and the above checks, it is 100% sure that the value exists, hence the phpstan-ignore instead of a default value.

Reply:
yes, i unterstand, but if it would be still great, if we can get rid of an ignore.
as alternative: use the array_filter only for the parameter of method hasAddressRequiredData, or within the method itself - then you should not get an phpstan error.

Reply:
I removed the array_filter because it was unnecessary anyway.